keys/keys.go

```diff
@@ -357,3 +357,17 @@ func MetaScanBounds(key proto.Key) (proto.Key, proto.Key) {
 	// Otherwise find the first entry greater than the given key in the same meta prefix.
 	return key.Next(), proto.Key(key[:len(Meta1Prefix)]).PrefixEnd()
 }
+
+// MetaReverseScanBounds returns the start and end keys of the range within which the
+// desired meta record can be found by means of an engine reverse scan. The given key
+// must be a valid RangeMetaKey as defined by ValidateRangeMetaKey.
+func MetaReverseScanBounds(key proto.Key) (proto.Key, proto.Key, error) {
+	if key.Equal(proto.KeyMin) || key.Equal(Meta1Prefix) {
+		return nil, nil, NewInvalidRangeMetaKeyError("KeyMin or Meta1Prefix can't be used as the key of reverse scan", key)
+	}
+	if key.Equal(Meta2Prefix) {
+		return Meta1Prefix, Meta1Prefix.PrefixEnd(), nil
+	}
+	// Otherwise find the first entry greater than the given key in the same meta prefix.
+	return key[:len(Meta1Prefix)], key.Next(), nil
```

Comment:
The rocksdb.Seek() will position the iterator at the first key that is >= "key". If we have ranges ["", "f") , ["f", "z") and your reverse scan range is `["a", "f")`, you will use `"f"` to look up the range. `"f"` is the key of the first meta row and the iterator will position it if we does not use `"f".Next()` .So the target range will be ignored by `Endkey` -exclusive .